hawc/apps/common/dynamic_forms/fields.py

```diff
@@ -113,9 +109,12 @@ class YesNoChoiceField(_Field):
     """Yes No field."""
 
     type: Literal["yes_no"] = "yes_no"
-    widget: Literal["radio_select"] = PydanticField("radio_select", const=True)
+    widget: Literal["radio_select"] = "radio_select"
 
-    choices: list[tuple[str, str]] = PydanticField([("yes", "Yes"), ("no", "No")], const=True)
+    choices: list[tuple[str, str]] = [
+        ("yes", "Yes"),
+        ("no", "No"),
+    ]  # TODO: make this act as Literal
```

Comment:
what does this comment mean? ok to remove?

Reply:
Is this what you're looking for?

```python
class YesNoChoiceField(_Field):
    """Yes No field."""

    type: Literal["yes_no"] = "yes_no"
    widget: Literal["radio_select"] = "radio_select"

    choices: list[tuple[Literal["yes", "no"], str]] = [("yes", "Yes"), ("no", "No")]

```

Seems like it works:

```python
from typing import Literal
x = Literal[2]
x = Literal[[2,3]]
x = Literal[["yes", "yes"],["no","no"]]
x
from pydantic import BaseModel, Field
class F(BaseModel):
    choices: list[tuple[Literal["yes", "no"], str]] = [("yes", "Yes"), ("no", "No")]
```

With a few tests:

```python
In [13]: x=F()

In [14]: x.choices
Out[14]: [('yes', 'Yes'), ('no', 'No')]

In [15]: F(choices=[('yes', 'Yes'), ('no', 'No')])
Out[15]: F(choices=[('yes', 'Yes'), ('no', 'No')])

In [16]: F(choices=[('yes', 'Yes'), ('no', 'zNo')])
Out[16]: F(choices=[('yes', 'Yes'), ('no', 'zNo')])

In [17]: F(choices=[('yes', 'Yezs'), ('no', 'zNo')])
Out[17]: F(choices=[('yes', 'Yezs'), ('no', 'zNo')])

In [18]: F(choices=[('yes', 'Yezs'), ('noz', 'zNo')])
--------------------------------------------------------------
ValidationError              Traceback (most recent call last)
Cell In[18], line 1
----> 1 F(choices=[('yes', 'Yezs'), ('noz', 'zNo')])

File ~/dev/hawc/venv/lib/python3.11/site-packages/pydantic/main.py:164, in BaseModel.__init__(__pydantic_self__, **data)
    162 # `__tracebackhide__` tells pytest and some other tools to omit this function from tracebacks
    163 __tracebackhide__ = True
--> 164 __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)

ValidationError: 1 validation error for F
choices.1.0
  Input should be 'yes' or 'no' [type=literal_error, input_value='noz', input_type=str]
    For further information visit https://errors.pydantic.dev/2.4/v/literal_error
```

Reply:
I did something like this in HERO too

Reply:
The way it was before was that if it was passed anything for choices besides `[('yes', 'Yes'), ('no', 'No')]` it would have failed, which I was not sure how to replicate strictly with typing (I think the closest we could get would be something like `tuple[tuple[Literal["yes"],Literal["Yes"]],tuple[Literal["no"],Literal["No"]]]`). Your proposed solution makes more sense behavior wise though, I see no reason that a user shouldn't be able to change the choice labels.